geonode/storage/manager.py

```diff
@@ -199,7 +205,9 @@ def copy_files_list(self, files: List[str]):
                     new_file = f"{new_path}/{suffixed_name}{ext}"
                 else:
                     new_file = f"{new_path}/{old_file_name}_{random_suffix}{ext}"
-                out.append(self.copy_single_file(open_file, new_file))
+                _new_path = self.copy_single_file(open_file, new_file)
+                out.append(_new_path)
+            os.chmod(_new_path, settings.FILE_UPLOAD_PERMISSIONS)
```

Comment:
I guess there's an indentation error here

Reply:
the cmod should be done once the file is closed before taking the next item in the loop

Reply:
but the `_new_path` variable is not defined there I guess, it's inside the `with` statement

Reply:
i add and additional check to verify that _new_path is filled before re-assign the permission. Is wanted to be done just before the loop so we can ensure that the file is closed before trying to access it